src/category_theory/limits/preserves/shapes/terminal.lean

```diff
@@ -0,0 +1,118 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+import category_theory.limits.preserves.limits
+import category_theory.limits.shapes
+
+/-!
+# Preserving terminal object
+
+Constructions to relate the notions of preserving terminal objects and reflecting terminal objects
+to concrete objects.
+
+In particular, we show that `terminal_comparison G` is an isomorphism iff `G` preserves terminal
+objects.
+-/
+
+universes v u₁ u₂
+
+noncomputable theory
+
+namespace category_theory
+open category limits
+
+variables {C : Type u₁} [category.{v} C]
+variables {D : Type u₂} [category.{v} D]
+variables (G : C ⥤ D)
+
+namespace preserves_terminal
+
+variables (X : C)
+
+/--
+The map of an empty cone is a limit iff the empty cone consisting of the mapped object is a limit.
+-/
+def terminal_map_cone_limit :
```

Comment:
Just as with the other PR, I think I would rather have a name like
```suggestion
def is_limit_map_cone_equiv :
```

Reply:
I agree it's good to standardise the names - let's discuss in that PR and once that's sorted I'll fix this one.